import {
  Code_Template,
  Url_Template,
  Welcome_Template,
} from "../../libs/verificationCodeTemplate.js";
import { transporter } from "./verificationEmailConfig.js";

// Middleware encargado de enviar el codigo de verificaci√≥n al correo
export const SendVerificationCode = async (email, verificationCode) => {
  try {
    await transporter.sendMail({
      from: '"GOBSUCRE" <emaildev1214@gmail.com>',
      to: email,
      subject: "C√≥digo de verificaci√≥n. ‚öô",
      text: "Gracias por registrarte GOBSUCRE. Por favor, verifica tu direcci√≥n de correo para continuar.",
      html: Code_Template.replace(
        "{verificationCode}",
        verificationCode
      ).replace(
        "{cabecera}",
        "Este es tu c√≥digo de verificaci√≥n verificaci√≥n."
      ),
      // html body
    });

    return {
      success: true,
      message:
        "Hemos enviado el c√≥digo de verificaci√≥n a tu correo. Por favor, revisa tu bandeja de entrada.",
    };
  } catch (error) {
    console.log(
      `No se ha podido enviar el c√≥digo de verificaci√≥n. Estamos trabajando para resolver este problema. ${error}`
    );

    return {
      success: false,
      message:
        "No se ha podido enviar el c√≥digo de verificaci√≥n. Estamos trabajando para resolver este problema.",
    };
  }
};

// Middleware encargado de enviar el url para restaurar la contrase√±a al correo
export const SendUrlResetPassword = async (email, resetPasswordUrl) => {
  try {
    await transporter.sendMail({
      from: '"GOBSUCRE" <emaildev1214@gmail.com>',
      to: email,
      subject: "Restablecer contrase√±a. ‚öô",
      text: "Hemos recibido una solicitud para restablecer tu contrase√±a en GOBSUCRE.",
      html: Url_Template.replace("{urlCode}", resetPasswordUrl).replace(
        "{cabecera}",
        "Link de reset de password."
      ),
      // html body
    });

    return {
      success: true,
      message:
        "Te hemos enviado un enlace para restablecer tu contrase√±a. Por favor, revisa tu bandeja de entrada y sigue las instrucciones.",
    };
  } catch (error) {
    console.log(
      `No se ha podido enviar el enlace para restablecer tu contrase√±a. Estamos trabajando para resolver este problema. ${error}`
    );

    return {
      success: false,
      message:
        "No se ha podido enviar el enlace para restablecer tu contrase√±a. Estamos trabajando para resolver este problema.",
    };
  }
};

// Middleware encargado de enviar un mensaje de bienvenida y confirmaci√≥n de registro
export const WelcomeEmail = async (email, name) => {
  try {
    await transporter.sendMail({
      from: '"GOBSUCRE" <emaildev1214@gmail.com>',
      to: email,
      subject: "Cuenta verificada con exito. ‚úî",
      text: "Bienvenido, tu cuenta ha sido verificada en GOBSUCRE.",
      html: Welcome_Template.replace("{name}", name),
    });

    return {
      success: true,
      message:
        "Hemos enviado el c√≥digo de verificaci√≥n a tu correo. Por favor, revisa tu bandeja de entrada.",
    };
  } catch (error) {
    console.log(
      `No se ha podido enviar el c√≥digo de verificaci√≥n. Estamos trabajando para resolver este problema. ${error}`
    );

    return {
      success: false,
      message:
        "Hemos enviado el c√≥digo de verificaci√≥n a tu correo. Por favor, revisa tu bandeja de entrada.",
    };
  }
};

// Middleware encargado de reenviar el codigo de verificaci√≥n al correo
export const ResendValidationCode = async (email, verificationCode) => {
  try {
    await transporter.sendMail({
      from: '"GOBSUCRE" <emaildev1214@gmail.com>',
      to: email,
      subject: "Hemos generado tu nuevo c√≥digo de verificaci√≥n. üéØ",
      text: "Usa este c√≥digo de verificaci√≥n para completar el proceso.",
      html: Code_Template.replace(
        "{verificationCode}",
        verificationCode
      ).replace("{cabecera}", "Hemos generado tu nuevo c√≥digo de validaci√≥n."),
    });

    return {
      success: true,
      message:
        "Hemos enviado el c√≥digo de validaci√≥n a tu correo. Por favor, revisa tu bandeja de entrada.",
    };
  } catch (error) {
    console.log(
      `No se ha podido enviar el c√≥digo de validaci√≥n. Estamos trabajando para resolver este problema. ${error}`
    );

    return {
      success: false,
      message:
        "No se ha podido enviar el c√≥digo de validaci√≥n. Estamos trabajando para resolver este problema.",
    };
  }
};
